api/src/main/java/org/openbaton/nfvo/api/runtime/RestNetworkServiceRecord.java

```diff
@@ -155,14 +158,25 @@ public NetworkServiceRecord create(
           BadFormatException, VimDriverException, QuotaExceededException, PluginException,
           MissingParameterException, BadRequestException {
 
-    log.debug("Json Body is" + jsonObject);
-    Type mapType = new TypeToken<Map<String, Configuration>>() {}.getType();
+    List keys = new LinkedList();
+    Map vduVimInstances = new HashMap();
+    Map<String, Configuration> configurations = new HashMap<>();
+
+    log.debug("Json Body is " + jsonObject);
+    if (jsonObject != null) {
+      if (jsonObject.has("keys")) {
+        keys = gson.fromJson(jsonObject.getAsJsonArray("keys"), List.class);
+      }
+      if (jsonObject.has("vduVimInstances")) {
+        vduVimInstances = gson.fromJson(jsonObject.getAsJsonObject("vduVimInstances"), Map.class);
+      }
+      if (jsonObject.has("configurations")) {
+        Type mapType = new TypeToken<Map<String, Configuration>>() {}.getType();
+        configurations = gson.fromJson(jsonObject.get("configurations"), mapType);
+      }
+    }
     return networkServiceRecordManagement.onboard(
-        id,
-        projectId,
-        gson.fromJson(jsonObject.getAsJsonArray("keys"), List.class),
-        gson.fromJson(jsonObject.getAsJsonObject("vduVimInstances"), Map.class),
-        (Map) gson.fromJson(jsonObject.get("configurations"), mapType));
+        id, projectId, keys, vduVimInstances, configurations);
```

Comment:
From the checks I did and the code the default parameters were passed. However since the checks later in the onboard functions are done against null, I think it would be better to keep the consistency and send null if nothing has been passed for a given parameter.